### PR TITLE
Add Hadamard operator in utils

### PR DESCRIPTION
**Pull request by abocquet**
Solves DYN-130

### CODE REVIEW

File: dynamiqs/utils/operators.py [hunk 1]
```diff
@@ -610,3 +611,43 @@ def sigmam(
...
     )
+
+
+def hadamard(
+    n: int = 1,
+    *,
+    dtype: torch.complex64 | torch.complex128 | None = None,
+    device: str | torch.device | None = None,
+) -> Tensor:
+    r"""Returns the Hadamard transform on `n` qubits
```
Comment (gautierronan) [inline dynamiqs/utils/operators.py:622]:
```suggestion
    r"""Returns the Hadamard transform on `n` qubits.
```

File: dynamiqs/utils/operators.py [hunk 2]
```diff
@@ -610,3 +611,43 @@ def sigmam(
...
+
+    It is defined by $H_0=1$ and
+    $$
+        H_m = \frac{1}{\sqrt2} \begin{pmatrix}
+            H_{m-1} & H_{m-1} \\
+            H_{m-1} & -H_{m-1}
+        \end{pmatrix}
+    $$
```
Comment (gautierronan) [inline dynamiqs/utils/operators.py:630]:
```suggestion
    For a single qubit, it is defined by
    $$
        H = \frac{1}{\sqrt2} \begin{pmatrix}
            1 & 1 \\
            1 & -1
        \end{pmatrix}
    $$
    
    For `n` qubits, it is defined by the tensor product of Hadamard matrices: 
    $$
        H_n = \bigotimes_{k=1}^n H
    $$
```

Comment (gautierronan) [reply dynamiqs/utils/operators.py:630]: A bit more canonical IMO.

File: dynamiqs/utils/operators.py [hunk 3]
```diff
@@ -610,3 +611,43 @@ def sigmam(
...
+
+    Args:
+        n: integer number of qubits to act on
```
Comment (gautierronan) [inline dynamiqs/utils/operators.py:633]:
```suggestion
        n: Number of qubits to act on.
```

**Review by gautierronan — CHANGES_REQUESTED** [review]
Just a few comments on the doc. Otherwise looks good :)

**Comment by abocquet** [comment]
Solves DYN-130

**Review by gautierronan — DISMISSED** [review]

**Comment by pierreguilmin** [comment]
> Solves DYN-130

You need to put it in the PR description for the linking with linear to work 😉

**Comment by pierreguilmin** [comment]
I'll try to review this soon.

**Review by gautierronan — APPROVED** [review]

**Comment by pierreguilmin** [comment]
Loots of very small changes, I dissected them into small commits, mostly conventions, I'll add these to my draft. I removed the `n=0` case, I'm not sure it makes sense to have it. I'll wait for your approval to merge @abocquet 😉

**Review by pierreguilmin — APPROVED** [review]

**Pull request merged**